### PR TITLE
fix failing e2e test

### PR DESCRIPTION
**Pull request by shiqizng**
## Summary
make e2e fails with nil pointer error with a new s3 artifact, 
```
output from 'go' 'run' 'cmd/e2equeries/main.go' '-pg' 'host=e2e-db port=5432 user=algorand password=algorand dbname=indexer_db sslmode=disable' '-q'
panic: runtime error: invalid memory address or nil pointer dereference
e2e_1     | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf94ac9]
```
This PR fixes this issue. 

## Test Plan
run make e2e to validate test completes without error.

### CODE REVIEW

File: misc/e2elive.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def main():
         s3 = boto3.client('s3', config=Config(signature_version=UNSIGNED))
         tarname = 'net_done.tar.bz2'
         tarpath = os.path.join(tempdir, tarname)
-        firstFromS3Prefix(s3, bucket, 'indexer/e2e3', tarname, outpath=tarpath)
+        firstFromS3Prefix(s3, bucket, 'indexer/e2e4', tarname, outpath=tarpath)
```
Comment (winder) [inline misc/e2elive.py:62]: This is also specified in `misc/buildtestdata.sh`

File: cmd/e2equeries/main.go [hunk 2]
```diff
@@ -44,7 +44,9 @@ func main() {
 	rowchan, _ := db.Transactions(context.Background(), rekeyTxnQuery)
 	for txnrow := range rowchan {
 		maybeFail(txnrow.Error, "err rekey txn %v\n", txnrow.Error)
-		rekeyedAuthAddrs = append(rekeyedAuthAddrs, txnrow.Txn.Txn.RekeyTo)
+		if txnrow.Txn != nil {
+			rekeyedAuthAddrs = append(rekeyedAuthAddrs, txnrow.Txn.Txn.RekeyTo)
+		}
```
Comment (winder) [inline cmd/e2equeries/main.go:49]: I'm not exactly sure whats happening here, but perhaps we need to check both
```suggestion
                t := txnrow.Txn
		if txnrow.RootTxn != nil {
		        t = txnrow.RootTxn
		}
		rekeyedAuthAddrs = append(rekeyedAuthAddrs, t.Txn.RekeyTo)
```

File: cmd/e2equeries/main.go [hunk 3]
```diff
@@ -63,7 +65,7 @@ func main() {
 	}
 	if !foundRekey {
 		// this will report the error in a handy way
-		printAccountQuery(db, idb.AccountQueryOptions{EqualToAuthAddr: rekeyedAuthAddrs[0][:], Limit: 1})
+		printAccountQuery(db, idb.AccountQueryOptions{EqualToAuthAddr: rekeyedAuthAddrs[0][:], Limit: 0})
```
Comment (winder) [inline cmd/e2equeries/main.go:68]: What error is this supposed to be reporting?

Comment (winder) [reply cmd/e2equeries/main.go:68]: I think I see it. This triggers the "came up short" error?

Comment (shiqizng) [reply cmd/e2equeries/main.go:68]: yes

**Review by winder — APPROVED** [review]

**Pull request merged**